providers/globus.go

```diff
@@ -0,0 +1,171 @@
+package providers
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/apis/sessions"
+)
+
+type GlobusProvider struct {
+	*ProviderData
+}
+
+type Token struct {
+	AccessToken string  `json:"access_token"`
+	ExpiresIn   int     `json:"expires_in"`
+	Scope       string  `json:"scope"`
+	Tokens      []Token `json:"other_tokens"`
+}
+
+//https://auth.globus.org/.well-known/openid-configuration
+func NewGlobusProvider(p *ProviderData) *GlobusProvider {
+	p.ProviderName = "Globus"
+	if p.LoginURL == nil || p.LoginURL.String() == "" {
+		p.LoginURL = &url.URL{
+			Scheme: "https",
+			Host:   "auth.globus.org",
+			Path:   "/v2/oauth2/authorize",
+		}
+	}
+	if p.RedeemURL == nil || p.RedeemURL.String() == "" {
+		p.RedeemURL = &url.URL{
+			Scheme: "https",
+			Host:   "auth.globus.org",
+			Path:   "/v2/oauth2/token",
+		}
+	}
+	if p.ValidateURL == nil || p.ValidateURL.String() == "" {
+		p.ValidateURL = &url.URL{
+			Scheme: "https",
+			Host:   "auth.globus.org",
+			Path:   "/v2/oauth2/token/introspect",
+		}
+	}
+	if p.ProfileURL == nil || p.ProfileURL.String() == "" {
+		p.ProfileURL = &url.URL{
+			Scheme: "https",
+			Host:   "auth.globus.org",
+			Path:   "/v2/oauth2/userinfo",
+		}
+	}
+	if p.Scope == "" {
+
+		// TODO:  scopes should be configurable
+		p.Scope = "openid email profile urn:globus:auth:scope:auth.globus.org:view_identities urn:globus:auth:scope:transfer.api.globus.org:all urn:globus:auth:scope:data.materialsdatafacility.org:all urn:globus:auth:scope:transfer.api.globus.org:all urn:globus:auth:scope:search.api.globus.org:search"
+
+	}
+	return &GlobusProvider{ProviderData: p}
+}
+
+func (p *GlobusProvider) GetEmailAddress(c context.Context, s *sessions.SessionState) (string, error) {
```

Comment:
This is a deprecated interface -- you should take a look at the other providers and how they use `EnrichSession` now instead of this.